### PR TITLE
CB-4961 Control files downloading ability

### PR DESCRIPTION
**Pull request by sergeyteleshev**

### CODE REVIEW

File: webapp/packages/plugin-data-viewer/src/DataViewerSettingsService.ts [hunk 1]
```diff
@@ -73,6 +78,7 @@ export class DataViewerSettingsService extends Dependency {
       createSettingsAliasResolver(this.settingsResolverService, this.settings, {
         'plugin.data-viewer.disableEdit': 'core.app.dataViewer.disableEdit',
         'plugin.data-viewer.disableCopyData': 'core.app.dataViewer.disableCopyData',
+        'plugin.data-viewer.disableDownload': 'core.app.dataViewer.disableDownload',
```
Comment (Wroud) [inline webapp/packages/plugin-data-viewer/src/DataViewerSettingsService.ts]: Please remove, aliases needed only for old settings, and it's not relevant for the new one

**Review by Wroud — CHANGES_REQUESTED** [review]

File: webapp/packages/plugin-data-viewer/src/DatabaseDataModel/Actions/ResultSet/ResultSetDataContentAction.ts [hunk 2]
```diff
@@ -42,6 +43,7 @@ export class ResultSetDataContentAction extends DatabaseDataAction<any, IDatabas
     private readonly graphQLService: GraphQLService,
     private readonly quotasService: QuotasService,
     private readonly cache: ResultSetCacheAction,
+    private readonly dataViewerService: DataViewerService,
```
Comment (Wroud) [inline webapp/packages/plugin-data-viewer/src/DatabaseDataModel/Actions/ResultSet/ResultSetDataContentAction.ts]: this is not responsibility of database data model - it presents only database not hight level API or logic

**Review by Wroud — CHANGES_REQUESTED** [review]

**Review by Wroud — APPROVED** [review]

**Comment by Wroud** [comment]
please fix unit tests

**Comment by sergeyteleshev** [comment]
I removed unit test cause moved the setting to `DataViewerSettingsService`
And cause it is here in cache it shows it as fallen

**Comment by sergeyteleshev** [comment]
Снимок экрана 2024-06-28 в 13 22 36

**Review by devnaumov — APPROVED** [review]

File: webapp/packages/plugin-data-viewer/src/ValuePanelPresentation/ImageValue/ImageValuePresentation.tsx [hunk 3]
```diff
@@ -57,11 +57,23 @@ export const ImageValuePresentation: TabContainerPanelComponent<IDataValuePanelP
       },
     );
 
+    function imageOnContextMenuHandler(event: React.MouseEvent<HTMLImageElement>) {
```
Comment (devnaumov) [inline webapp/packages/plugin-data-viewer/src/ValuePanelPresentation/ImageValue/ImageValuePresentation.tsx]: Maybe onImageContextMenu or imageContextMenuHandler

File: webapp/packages/plugin-data-viewer/src/DataViewerSettingsService.ts [hunk 4]
```diff
@@ -45,6 +46,10 @@ export class DataViewerSettingsService extends Dependency {
     return this.settings.getValue('plugin.data-viewer.disableCopyData');
   }
 
+  get disableExportData(): boolean {
+    return this.settings.getValue('plugin.data-export.disabled');
+  }
```
Comment (Wroud) [inline webapp/packages/plugin-data-viewer/src/DataViewerSettingsService.ts]: please add old alias:
```
      /** @deprecated Use settings instead, will be removed in 23.0.0 */
      createSettingsAliasResolver(this.settingsResolverService, this.settings, { 'plugin.data-export.disabled': 'plugin_data_export.disabled' }),
```
also rename `plugin.data-export.disabled` to `plugin.data-viewer.export.disabled` and add alias to `plugin.data-export.disabled`

File: webapp/packages/plugin-data-viewer/src/DataViewerSettingsService.ts [hunk 5]
```diff
@@ -147,6 +152,16 @@ export class DataViewerSettingsService extends Dependency {
           description: 'settings_data_editor_fetch_max_description',
           group: DATA_EDITOR_SETTINGS_GROUP,
         },
+        {
+          group: DATA_EDITOR_SETTINGS_GROUP,
+          key: 'plugin.data-export.disabled',
+          type: ESettingsValueType.Checkbox,
+          name: 'settings_data_editor_disable_data_download_name',
+          description: 'settings_data_editor_disable_data_download_description',
```
Comment (Wroud) [inline webapp/packages/plugin-data-viewer/src/DataViewerSettingsService.ts]: please rename from download to export

File: webapp/packages/plugin-data-viewer/src/locales/de.ts [hunk 6]
```diff
@@ -41,6 +41,8 @@ export default [
   ['settings_data_editor_disable_edit_description', 'Deaktivieren Sie die Bearbeitung von Daten in Data Viewer für Nicht-Admin-Benutzer'],
   ['settings_data_editor_disable_data_copy_name', 'Kopie deaktivieren'],
   ['settings_data_editor_disable_data_copy_description', 'Deaktivieren Sie das Kopieren von Daten in Data Viewer für Nicht-Admin-Benutzer'],
+  ['settings_data_editor_disable_data_download_name', 'Disable Export'],
+  ['settings_data_editor_disable_data_download_description', 'Disable exporting of data in Data Viewer for non-admin users'],
```
Comment (Wroud) [inline webapp/packages/plugin-data-viewer/src/locales/de.ts]: please rename to Export

**Review by Wroud — CHANGES_REQUESTED** [review]

**Comment by Wroud** [comment]
please fix tests

**Review by Wroud — APPROVED** [review]

**Review by Wroud — APPROVED** [review]

**Pull request merged**